packages/horizon/contracts/staking/HorizonStaking.sol

```diff
@@ -278,6 +278,8 @@ contract HorizonStaking is HorizonStakingBase, IHorizonStakingMain {
         uint256 tokens
     ) external override notPaused {
         require(tokens != 0, HorizonStakingInvalidZeroTokens());
+        require(msg.sender == verifier || msg.sender == address(_graphPayments()), HorizonStakingInvalidDelegationPoolSender(msg.sender));
```

Comment:
I think this makes sense but worth discussing with OZ just in case, it could increase the risk of rounding attacks etc.

(If we do go ahead as it is, let's rename the PR without the "access control" part)

Reply:
I've renamed and will let OZ know so they can give their feedback.